### ISSUE TITLE
Some questions for the score matrix generating process in the single-cell competition

### ISSUE THREAD

**Issue by HelloWorldLTY** [gao-lab/GLUE#3]
Hello, I am Tianyu. Thanks for your great presentation in the single-cell competition workshop. After going through your code, I am a little confused about the method you used to generate the score matrix, I have two questions and I am looking forward to your reply.

1. I think the input of the snn_matching function should be something like the embeddings data of different modalities, is it correct? Are these embeddings same as the u1,u2 you used in the CLUE? Thanks.
2. I wonder why you don't choose cosine similarity or gaussian kernel to process the value in the matrix but choose to use Jaccard value. Is it very common metric used for evaluating the similarity of different vectors? I think we can use this value in set comparison. 

Thanks a lot!

**Comment by Jeff1995** [comment]
Hi Tianyu!

1. The answer is yes for the first question. The input embeddings are just u1 and u2.
2. For the second question, we did use Gaussian kernel at first. We found that the matching score was highest when kernel width was very small (effectively concentrating all the score into the nearest neighbor).

   SNN is quite commonly used in scRNA-seq clustering, so we also gave it a try. The difference in matching score between a small Gaussian kernel and the SNN Jaccard with k=1 was actually fairly small, on the scale of 0.001-0.003. We opted for SNN because there is no distance-dependent hyperparameter like the kernel width, so it should be more robust to scaling in the embedding space.

   As for cosine similarity, we never tried that, but given the observation that the matching score goes higher when all scores are concentrated at the nearest neighbor, it probably won't work well because it is generally much smoother.
